dev/NumberBox/NumberBox.cpp

```diff
@@ -0,0 +1,555 @@
+    }
+
+    // MinMaxMode Wrapping
+    if (MinMaxMode() == winrt::NumberBoxMinMaxMode::WrapEnabled && GetBoundState(newVal) != BoundState::InBounds)
+    {
+        while (newVal > MaxValue())
+        {
+            newVal = MinValue() + (newVal - MaxValue()) - 1;
+        }
+        while (newVal < MinValue())
+        {
+            newVal = MaxValue() - abs(newVal - MinValue()) + 1;
+        }
+    }
+    // Input Overwriting - Coerce to min or max
+    else if (BasicValidationMode() == winrt::NumberBoxBasicValidationMode::InvalidInputOverwritten && GetBoundState(newVal) != BoundState::InBounds)
+    {
+        if (newVal > MaxValue() && (MinMaxMode() == winrt::NumberBoxMinMaxMode::MaxEnabled || MinMaxMode() == winrt::NumberBoxMinMaxMode::MinAndMaxEnabled))
+        {
+            newVal = MaxValue();
+        }
+        else if (newVal < MinValue() && (MinMaxMode() == winrt::NumberBoxMinMaxMode::MinEnabled || MinMaxMode() == winrt::NumberBoxMinMaxMode::MinAndMaxEnabled))
+        {
+            newVal = MinValue();
+        }
+    }
+    
+    // Safeguard for floating point imprecision errors
+    int StepFreqSigDigits = ComputePrecisionRounderSigDigits(newVal);
+    m_stepPrecisionRounder.SignificantDigits(StepFreqSigDigits);
+    newVal = m_stepPrecisionRounder.RoundDouble(newVal);
+
+    // Update Text and Revalidate new value
+    Value(newVal);
+    UpdateTextToValue();
+    ValidateInput();
+}
+
+
+// Check if text resembles formulaic input to determine if parser should be executed
+bool NumberBox::IsFormulaic(const winrt::hstring& in)
+{
+    std::wstring input(in);
+    std::wregex formula(L"^([0-9()\\s]*[+-/*^%]+[0-9()\\s]*)+$");
+    std::wregex negval(L"^-([0-9])+(.([0-9])+)?$");
+    return (std::regex_match(input, formula) && !std::regex_match(input, negval));
+}
+
+// Computes the number of significant digits that precision rounder should use. This helps to prevent floating point imprecision errors. 
+int NumberBox::ComputePrecisionRounderSigDigits(double newVal)
+{
+    auto const oldVal = Value();
+    auto const stepFreq = StepFrequency();
+
+    // Run formatter on both values to discard trailing and leading 0's.
+    std::wstring formattedVal(m_stepPrecisionFormatter.Format(oldVal));
+    std::wstring formattedStep(m_stepPrecisionFormatter.Format(stepFreq));
+    std::wstring formattedNew(m_stepPrecisionFormatter.Format(newVal));
+
+    // Get size of only decimal portion of both old numbers. 
+    int oldValSig = static_cast<int>(formattedVal.substr(formattedVal.find_first_of('.') + 1).size());
+    int StepSig = static_cast<int>(formattedStep.substr(formattedStep.find_first_of('.') + 1).size());
+
+    // Pick bigger of two decimal sigDigits
+    int result = std::max(oldValSig, StepSig);
+    // append # of integer digits from new value
+    result += (int) formattedNew.substr(0, formattedNew.find_first_of('.')).size();
+    return result;
+}
+
+void NumberBox::NormalizeShorthandOperations()
```

Comment:
This function deserves a comment. I didn't even know this was a feature until you demoed it!